src/modules/mc_pos_control/Takeoff/Takeoff.cpp

```diff
@@ -115,6 +116,10 @@ float Takeoff::updateRamp(const float dt, const float takeoff_desired_vz)
 	}
 
 	if (_takeoff_state == TakeoffState::rampup) {
+		if (!PX4_ISFINITE(takeoff_desired_vz)) {
+			return _takeoff_ramp_vz;
```

Comment:
I think the ramp should continue to the maximum speed constraint to avoid getting stuck in case `takeoff_desired_vz` stays always NaN. I can make a suggestion.

Reply:
Here's the suggestion: https://github.com/PX4/Firmware/pull/14339